File: 3501-delete-nodes-from-linked-list-present-in-array/delete-nodes-from-linked-list-present-in-array.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def modifiedList(self, nums: List[int], head: Optional[ListNode]) -> Optional[ListNode]:
        numSet = set()
        numSet.update(nums)

        while head and head.val in numSet:
            head = head.next
        
        temp = head
        while temp:
            while temp.next and temp.next.val in numSet:
                temp.next = temp.next.next if temp.next.next else None
            temp = temp.next

        return head

File: 3501-delete-nodes-from-linked-list-present-in-array/delete-nodes-from-linked-list-present-in-array.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode modifiedList(int[] nums, ListNode head) {
        Arrays.sort(nums);

        while (null!=head && Arrays.binarySearch(nums,head.val)>=0){
            head = head.next;
        }

        ListNode temp = head;
        while (null!=temp){
            while (null!=temp.next && Arrays.binarySearch(nums,temp.next.val)>=0){
                temp.next = temp.next.next;
            }
            temp = temp.next;
        }
    
        return head;
    }
}